protocol/parser/parser.go

```diff
@@ -107,6 +107,27 @@ func ParseBlockFromParams(rpcInput RPCInput, blockParser spectypes.BlockParser)
 	return rpcInput.ParseBlock(resString)
 }
 
+func ParseSpecIDFromReply(rpcInput RPCInput, blockParser spectypes.BlockParser) (string, error) {
```

Comment:
duplicated code in ParseBlockFromReply, you can call ParseSpecIDFromReply -> ParseFromReply and use it inside ParseBlockFromReply as they are basically the same except for the return statement